### ISSUE TITLE
Add rendering for wall=no

### ISSUE THREAD

**Issue by bagage** [gravitystorm/openstreetmap-carto#2623]
`wall=no` should be used for `slight construction like a shed or balcony opened on one side at least or without foundation`. It is [mainly used](https://taginfo.openstreetmap.org/tags/wall=no#map) in France due to cadastre import:

![image](https://cloud.githubusercontent.com/assets/1451988/25580576/7231c80e-2e82-11e7-9a7c-aeeb605b8f99.png)

In OSM French style, it is [rendered with lower opacity](https://github.com/cquest/osmfr-cartocss/blob/master/buildings.mss#L44-L48) starting at zoom 18:

![image](https://cloud.githubusercontent.com/assets/1451988/25580645/c89f5026-2e82-11e7-9fb7-7d59c5e53840.png)![image](https://cloud.githubusercontent.com/assets/1451988/25580650/d20e88d4-2e82-11e7-999c-206b4db7bcd2.png)

**Comment by mboeringa** [comment]
**wall=no + building=yes** to denote one-sided open shed?

Honestly, I really think this kind of stuff simply classifies as **building=shed**, and should have been tagged and imported as such by the french community (there are some wacko import taggings in the Netherlands where I live too by the way... ).

The fact that the shed is open on one side, could have been tagged differently and more logical and explicit than this. Maybe **open_sided=yes**?

**Comment by nebulon42** [comment]
I also think `wall=no` is rather ill-defined.

**Comment by mboeringa** [comment]
Even more reason to tag with **building=shed** is that most of these are probably firewood sheds:
firewood shed:

https://www.google.nl/search?q=firewood&ie=utf-8&oe=utf-8&client=firefox-b&gws_rd=cr&ei=b0kHWfuJC4iOgAa75JqQBg#q=firewood+shed

which seem to be called like that in English, considering the above Google search...

Maybe another tagging could be:
**building=shed + content=fire_wood**

**Comment by bagage** [comment]
At least in France, **wall=no + building=yes** can either refer to one of these:

Part of a building (on the right side here):
![image](https://cloud.githubusercontent.com/assets/1451988/25582434/9bc4f29e-2e8d-11e7-9de3-a9cc6b64811f.png)
Roof only structure:
![image](https://cloud.githubusercontent.com/assets/1451988/25582428/97b33b52-2e8d-11e7-9a83-88f3e3397e7c.png)

I don't know the history about this coice, but if you think it's misused or could be done better, I'll ask upstream :).

**Comment by mboeringa** [comment]
> I don't know the history about this coice, but if you think it's misused or could be done better, I'll ask upstream :).

The first picture seems to me nothing but a luxury version of **building=shed**, while the second is clearly **building=roof**. If no distinction can be made, I guess **building=shed** is the most logical choice to re-tag them, as **building=roof** is more often used and associated with bigger structures like roofs on railway or petrol stations, and not used for storage, like a shed.

**Comment by dieterdreist** [comment]
2017-05-01 16:50 GMT+02:00 mboeringa <notifications@github.com>:

> The first picture seems to me nothing but a luxury version of
> *building=shed*, while the second is clearly *building=roof*. If no
> distinction can be made, I guess *building=shed* is the most logical
> choice to re-tag them, as *building=roof* is more often used and
> associated with bigger structures like roofs on railway or petrol stations,
> and not used for storage, like a shed.
>



+1, there is documentation for wall=* as a subtag for barrier=wall, where
wall is used to provide details about the wall, even the French version
doesn't mention "no" as value (although it is very common due to your
import). https://wiki.openstreetmap.org/wiki/Key:wall
I would agree with building=* and wall=no being a synonym for a
building=roof with no walls, but in the case of at least one wall it seems
wrong (no is no, not?)

**Comment by bagage** [comment]
Just to be clear and ensure that we all are on the same page. 
France buildings are semi auto-imported from the French land register / cadastre. 
It contains almost all French buildings position **and** one single extra information: if the building is of type "heavy" or "light". 
"light" may basically means one of the following:
![out](https://user-images.githubusercontent.com/1451988/26844978-76cb152a-4af5-11e7-8227-7677413c5673.jpg)


So I don't think any building=* value will match all these cases simultaneously. 
On the other hand, dropping this information totally would be regrettable. This is the reason why another tag (`wall=no`) is used.
In the mean time, this information is actually not rendered while it could be useful. This is somewhat relative to symbol 526.2 "Canopy" / 852 Building pass-through used in orienteering, [which interestingly has no equivalent in OSM](https://wiki.openstreetmap.org/wiki/IOF_mapping#Man-made_features):

![capture d ecran de 2017-06-06 20-20-30](https://user-images.githubusercontent.com/1451988/26845093-c6b17a98-4af5-11e7-892d-8497e9753b7d.png)

**Comment by mboeringa** [comment]
I really think rendering of this tag is a dead end. I think the French community should reconsider what they want to do with this information:

1) - Revert this part of the building import and start from scratch mapping this when needed with more appropriate tags
2) - Keep it as is and accept unlikely support in a lot of renders
3) - Start enhancing the data by classifying it using more useful tags by field or imagery observation, e.g.:
* building=shed (+ content=firewood in specific cases?)
* building=roof
* building=greenhouse
* amenity=parking + building=yes
* whatever else seems appropriate and has tag and rendering support.

Personally, I'd go for 3)... The particular wildly varying mixture of buildings and structures you are showing for a "light" building, also strongly points in the need to enhance tagging using more appropriate tags and field observations.

**Comment by dieterdreist** [comment]
> On 6. Jun 2017, at 22:27, mboeringa <notifications@github.com> wrote:
> 
> Start enhancing the data by classifying it using more useful tags by field or imagery observation


+1, just because the cadastre didn't provide more building detail doesn't mean France has to remain at that level, you can always improve this first version with more precise tagging from survey

**Comment by bagage** [comment]
Yes obviously that's what we must target. But in the mean time should we just ignore this information? Why rendering `building=yes` in that case, since this is the same issue?
We all agree that `building=yes wall=no` is some generic tagging which must be improved, but I just wanted to have a better map visibility until we reach the "perfect mapping" of descripting every single building with proper `building=*` type.
If it doesn't make sense for you… well I cannot argue more actually :).

**Comment by HolgerJeromin** [comment]
Just to remember: osm-carto does not render any light buildings [different](#2532) then building=yes.
If we decide to render them (roof? garage? shed?) we could decide to include french light buildings, too.

But this style is (also) a mapper feedback loop not (only) to have a nice looking map. If osm-carto would render french-light building special who would stand up and clean up the import?
For example I never bothered about special building tags for residential buildings but tagged all =yes. No i have an app (streetcomplete) which asks building:level and roof questions for building=house/detached... 
=> i tagged a few thousands in my hometown.

**Comment by matkoniecz** [comment]
I am not convinced that supporting and encouraging this tag is a wise choice. If light/minor/shedlike building are to be rendered differently I think that building=* value should be used.

I plan on closing it per @mboeringa comments.
